core/model/src/model_geometry.cpp

```diff
@@ -238,6 +244,14 @@ void ModelGeometry::importSampledFieldGeometry(const libsbml::Model *model) {
   image = gsf.image.convertToFormat(QImage::Format_Indexed8);
   hasImage = true;
   pixelWidth = calculatePixelWidth(image.size(), physicalSize);
+
+  if (!sbmlAnnotation->compartmentColours.size()) {
```

Comment:
I think this should always be done: we want to replace whatever compartmentColours are in settings with the compartmentColours we just imported